compiler/generator_go.c

```diff
@@ -0,0 +1,1253 @@
+    g->I[0] = n;
+    w(g, "~Mbreak lab~I0~N");
+}
+
+static void write_failure(struct generator * g) {
+
+    if (str_len(g->failure_str) != 0) {
+        write_margin(g);
+        write_str(g, g->failure_str);
+        write_newline(g);
+    }
+    switch (g->failure_label)
+    {
+        case x_return:
+            w(g, "~Mreturn false~N");
+            g->unreachable = true;
+            break;
+        default:
+            g->I[0] = g->failure_label;
+            w(g, "~Mbreak lab~I0~N");
+            g->unreachable = true;
+    }
+}
+
+static void write_failure_if(struct generator * g, char * s, struct node * p) {
+
+    writef(g, "~Mif ", p);
+    writef(g, s, p);
+    writef(g, " ", p);
+    write_block_start(g);
+    write_failure(g);
+    write_block_end(g);
+    g->unreachable = false;
+}
+
+/* if at limit fail */
+static void write_check_limit(struct generator * g, struct node * p) {
+
+    if (p->mode == m_forward) {
+        write_failure_if(g, "env.Cursor >= env.Limit", p);
+    } else {
+        write_failure_if(g, "env.Cursor <= env.LimitBackward", p);
+    }
+}
+
+/* Formatted write. */
+static void writef(struct generator * g, const char * input, struct node * p) {
+    int i = 0;
+    int l = strlen(input);
+
+    while (i < l) {
+        int ch = input[i++];
+        if (ch != '~') {
+            write_char(g, ch);
+            continue;
+        }
+        switch (input[i++]) {
+            default: write_char(g, input[i - 1]); continue;
+            case 'C': write_comment(g, p); continue;
+            case 'f': write_block_start(g);
+                      write_failure(g);
+                      g->unreachable = false;
+                      write_block_end(g);
+                      continue;
+            case 'M': write_margin(g); continue;
+            case 'N': write_newline(g); continue;
+            case '{': write_block_start(g); continue;
+            case '}': write_block_end(g); continue;
+            case 'S': write_string(g, g->S[input[i++] - '0']); continue;
+            case 'B': write_b(g, g->B[input[i++] - '0']); continue;
+            case 'I': write_int(g, g->I[input[i++] - '0']); continue;
+            case 'V': write_varref(g, g->V[input[i++] - '0']); continue;
+            case 'W': write_varname(g, g->V[input[i++] - '0']); continue;
+            case 'L': write_literal_string(g, g->L[input[i++] - '0']); continue;
+            case '+': g->margin++; continue;
+            case '-': g->margin--; continue;
+            case 'n': write_string(g, g->options->name); continue;
+        }
+    }
+}
+
+static void w(struct generator * g, const char * s) {
+    writef(g, s, 0);
+}
+
+static void generate_AE(struct generator * g, struct node * p) {
+    const char * s;
+    switch (p->type) {
+        case c_name:
+            write_varref(g, p->name); break;
+        case c_number:
+            write_int(g, p->number); break;
+        case c_maxint:
+            write_string(g, "1<<32 - 1"); break;
```

Comment:
The operator precedence in Go appears to be different to other languages, so explicit parentheses seem wise if only to avoid confusing people who aren't familiar with Go.

But the expression seems wrong anyway - snowball integers should map to a signed integer type, so this should be the largest value of that signed integer type, but the current expression would be the largest value of an unsigned integer type.